Adding access mode info in storage classes (#837)
opendatahub-io/opendatahub-documentation: modules/about-storage-classes.adoc

When adding cluster storage to your project or workbench, you can choose from these enabled storage classes and access modes.

== Access modes

Storage classes create persistent volumes that can support different access modes, depending on the storage backend. Access modes control how a volume can be mounted and used by one or more workbenches. If a storage class allows more than one access mode, you can select the one that best fits your needs when you request storage. All persistent volumes support `ReadWriteOnce (RWO)` by default.

[cols="1,3"]
|===
|Access mode | Description

|`ReadWriteOnce (RWO)`
|The storage can be attached to a single workbench (or pod) at a time. This is the default and safest option for most individual workloads. `RWO` is always enabled by default and cannot be disabled by the administrator. 

|`ReadWriteMany (RWX)`

> |The storage can be attached to many workbenches simultaneously. This enables collaboration and shared data access but includes additional risks.

Author: Great point! Some of the potential risks are covered later in the module as a bulleted list, specifically wrt using the shared storage `RWX` mode.

|`ReadOnlyMany (ROX)`
|The storage can be attached to many workbenches as read-only. This is useful for sharing reference data without allowing changes.

|`ReadWriteOncePod (RWOP)`

> |The storage can be attached to a single pod on a single node as read-write. This is similar to RWO but with additional node-level restrictions.

Reviewer: ```suggestion
|The storage can be attached to a single pod on a single node with read-write permissions. `RWOP` is similar to `RWO` but includes additional node-level restrictions.
```

|===

=== Using shared storage (RWX)

The `ReadWriteMany (RWX)` access mode allows multiple workbenches to access and write to the same storage volume at the same time. This is useful for collaborative work where multiple users need to access shared datasets or project files.

However, shared storage introduces several risks:

> * *Data corruption*: If multiple workbenches try to change the same part of a file at the same time without proper coordination, the data can become corrupted or overwritten. Ensure your applications or workflows are designed to safely handle shared access, for example by using file locking or database transactions.

Reviewer: ```suggestion
* *Data corruption or data loss*: If multiple workbenches modify the same part of a file simultaneously, the data can become corrupted or lost. Ensure your applications or workflows are designed to safely handle shared access, for example, by using file locking or database transactions.
```

* *Accidental data loss*: One user could unintentionally delete or overwrite files that others are using. Always use caution when modifying shared data.

> * *Security and privacy*: If a workbench with access to shared storage is compromised, all data on that volume may be at risk. Only share sensitive data with trusted workbenches and users.

Reviewer 2: replace `may` with `might`

> To use shared storage safely, consider the following best practices:

Reviewer: ```suggestion
To use shared storage safely:
```

* *Know your application*: Ensure your tools or workflows are designed to work with shared storage and can manage simultaneous writes (for example, databases or distributed data processing frameworks).
* *Be cautious with changes*: Deleting or editing files affects everyone who shares the volume.
* *Back up your work*: Regular backups can help prevent data loss due to mistakes or misconfigurations.

> * *Limit access*: Only share RWX volumes with trusted users and secure workbenches.

Reviewer: ```suggestion
* Limit access to RWX volumes to trusted users and secure workbenches.
```

> * *Consider alternative access modes*: For most individual tasks, `ReadWriteOnce (RWO)` is safer because only one workbench can write to the volume at a time. Use `ReadWriteMany (RWX)` only when collaboration on a shared volume is required.

Reviewer: ```suggestion
*  Use `ReadWriteMany (RWX)` only when collaboration on a shared volume is required. For most individual tasks, `ReadWriteOnce (RWO)` is ideal because only one workbench can write to the volume at a time.
```
